### PR TITLE
Added memos

### PR DESCRIPTION
**Pull request by highghlow**
These apps are going to a a PITA to maintain... I should probably automate this.

# App Submission

### App name
Memos

### 256x256 SVG icon
_(Submit an icon with no rounded corners as it will be dynamically rounded with CSS. GitHub doesn't allow uploading SVGs directly, so please upload your icon to an alternate service, like https://svgur.com, and paste the link below.)_
_We will help finalize this icon before the app goes live in the Umbrel App Store._

Unfortunately this can't be made into an svg

https://www.usememos.com/logo.png

### Gallery images
_(Upload 3 to 5 high-quality gallery images (1440x900px) of your app in PNG format, or just upload 3 to 5 screenshots of your app and we'll help you design the gallery images.)_
_We will help finalize these images before the app goes live in the Umbrel App Store._

68747470733a2f2f7777772e7573656d656d6f732e636f6d2f64656d6f2e706e67

memos-app-user-interface-showing-a-picture-of-dogs-plus-a-note

### I have tested my app on:
- [ ] umbrelOS on a Raspberry Pi
- [ ] umbrelOS on an Umbrel Home
- [x] umbrelOS on Linux VM

### CODE REVIEW

**Comment by highghlow** [comment]
App request #992

File: memos/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    environment:
+      # <app-id>_<web-container-name>_1
+      # e.g. 'btc-rpc-explorer_web_1'
+      # Note that the '_1' at the end is needed
+      APP_HOST: memos_main_1
+      APP_PORT: 5230
+      PROXY_AUTH_WHITELIST: "/api/*"
+
+  main:
+    image: neosmemo/memos:0.21
```
Comment (nmfretz) [inline memos/docker-compose.yml:14]: Let's lock this in with a specific multi-arch digest so all users get the same image (even across app restarts):

```
$ sudo docker pull neosmemo/memos:0.21
0.21: Pulling from neosmemo/memos...
Digest: sha256:1d295e66cc2e75f1645872fc4a372f8f58efad17902f432758762d747984d6fb
Status: Downloaded newer image for neosmemo/memos:0.21
docker.io/neosmemo/memos:0.21
```

File: memos/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    restart: on-failure
+    stop_grace_period: 1m
+    volumes:
+      - ${APP_DATA_DIR}:/var/opt/memos
```
Comment (nmfretz) [inline memos/docker-compose.yml:18]: It's kinda confusing, but the ${APP_DATA_DIR} variable is `<path-to-umbrel-install>/app-data/<app-id>`, so to bind mount this to `<path-to-umbrel-install>/app-data/<app-id>/data` we need to change to:

```suggestion
    volumes:
      - ${APP_DATA_DIR}/data:/var/opt/memos
```

File: memos/umbrel-app.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+manifestVersion: 1
+id: memos
+category: files
+name: Memos
+version: "0.21"
+tagline: Easily capture and share your great thoughts.
```
Comment (nmfretz) [inline memos/umbrel-app.yml:6]: To stay consistent with the rest of the apps, let's remove the period at the end of the tagline.

File: memos/umbrel-app.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  chore\: fix typo by @sunxunle in #3106
+  feat\: add shortcut to edit the previous memo by @Dubzer in #3122
+  removed sanitization logic by @YohannKovacs in #3147
+  chore\: update Vietnamese translation by @ntheanh201 in #3155
+  fix\: link embedded content to uid instead of name by @ntheanh201 in #3164
+  feat\: handle restore memo when is in archived memo detail page by @ntheanh201 in #3165
+  cmd\: use the built-in slices library by @stayweek in #3172
+  chore\: fix memo link by @ntheanh201 in #3173
+  Set navigation icons to not shrink by @imikod in #3186
+  improve localization by @imikod in #3190
```
Comment (nmfretz) [inline memos/umbrel-app.yml:38]: We can remove release notes for the initial release on the app store:

`releaseNotes: ""`

File: memos/umbrel-app.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+  Open source and free forever. Embrace a future where creativity knows no boundaries with our open-source solution – free today, tomorrow, and always.
+
+  Self-hosting with Docker in just seconds. Enjoy the flexibility, scalability, and ease of setup that Docker provides, allowing you to have full control over your data and privacy.
+
+  Pure text with added Markdown support. Say goodbye to the overwhelming mental burden of rich formatting and embrace a minimalist approach.
+
+  Customize and share your notes effortlessly. With our intuitive sharing features, you can easily collaborate and distribute your notes with others.
+
+  RESTful API for third-party services. Embrace the power of integration and unleash new possibilities with our RESTful API support.
```
Comment (nmfretz) [inline memos/umbrel-app.yml:19]: Creating yaml that renders nicely in the app store UI is a bit finicky. One newline between paragraphs places each paragraph on a separate line, but with no whitespace between. Like this:

image

To get whitespace between each paragraph you can add **2** newlines between each paragraph. Alternatively, you can make each of the paragraphs a bullet point with a single newline between each to render nice bullet points like this:

```
description: >-
  A privacy-first, lightweight note-taking service.
  Easily capture and share your great thoughts.
  
  - Open source and free forever. Embrace a future where creativity knows no boundaries with our open-source solution – free today, tomorrow, and always.
  
  - Self-hosting with Docker in just seconds. Enjoy the flexibility, scalability, and ease of setup that Docker provides, allowing you to have full control over your data and privacy.
  
  - etc...
```

image

**Review by nmfretz — COMMENTED** [review]
Excellent, thanks for this one as well @highghlow 🙌! Please take a look at my suggested changes and then I will test and get gallery images made up.

File: memos/docker-compose.yml (hunk 1 above)
Comment (highghlow) [reply memos/docker-compose.yml]: Well, it turns out I forgot about the digest in every submission. I've fixed it.

File: memos/umbrel-app.yml (hunk 5 above)
Comment (highghlow) [reply memos/umbrel-app.yml:19]: I used bullet points. There could be an issue with the `-` character, but I checked that it parses.

**Comment by highghlow** [comment]
Fixed everything here. I'lll also go ahead and fix these issues in my other submissions.

**Comment by nmfretz** [comment]
> Fixed everything here. I'lll also go ahead and fix these issues in my other submissions.

Nice, thanks for being so on the ball and fixing the other submissions. Super appreciated.

We're getting gallery assets together now and will launch this when they are ready!

**Comment by nmfretz** [comment]
Thanks again for submitting Memos @highghlow. 

Gallery assets: https://github.com/getumbrel/umbrel-apps-gallery/tree/master/memos

I have pushed some final commits to finish things up, including changing the memos container user to umbrel (1000) instead of root.

Going live!
image

**Comment by Niclassslua** [comment]
image

<br><br>

The Memos icon is very low quality.

I would suggest using the one I uploaded. It's 900x900.

<br><br>

memos

**Pull request merged**